core/src/autogluon/core/searcher/local_grid_searcher.py

```diff
@@ -26,12 +39,27 @@ def _get_params_space(self) -> dict:
         param_space = dict()
         for key, val in self.search_space.items():
             if isinstance(val, Space):
-                if not isinstance(val, Categorical):
+                samples_num = self._get_samples_number(key)
+                if isinstance(val, Int):
+                    samples = min(val.upper - val.lower + 1, samples_num)
+                    param_space[key] = np.linspace(val.lower, val.upper, samples, dtype=int)
+                elif isinstance(val, Real):
+                    space = np.geomspace if val.log else np.linspace
+                    param_space[key] = space(val.lower, val.upper, num=samples_num)
```

Comment:
idk if its worth addressing but now this searcher doesn't respect default values on the first iteration. Worth mentioning in documentation that grid with Int/Real ignores 'default' values in search spaces.

Reply:
Added comment